crates/stages/src/sets.rs

```diff
@@ -129,8 +128,8 @@ where
     B: BodyDownloader + 'static,
     EF: ExecutorFactory,
 {
-    fn builder(self) -> StageSetBuilder<DB> {
-        Self::add_offline_stages(self.online.builder(), self.executor_factory)
+    fn builder(self) -> Result<StageSetBuilder<DB>, StageError> {
+        Self::add_offline_stages(self.online.builder()?, self.executor_factory)
```

Comment:
all of these result changes can be attributed to the OnlineStageSet

I believe these changes are not necessary if the OnlineStageset already captures the stages directly and doesn't create them on `build`